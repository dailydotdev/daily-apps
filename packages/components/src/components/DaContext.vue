<template>
  <VueContext class="context shadow2" ref="context"
              @close="$emit('close')" @open="onOpen"
              :close-on-click="false">
    <slot></slot>
  </VueContext>
</template>

<script>
import { VueContext } from 'vue-context';

export default {
  name: 'DaContext',
  components: { VueContext },

  methods: {
    open(event, data) {
      event.stopPropagation();
      setTimeout(() => this.$refs.context.open(event, data));
    },
    close() {
      this.$refs.context.close();
    },
    positionMenu({
      top, bottom, left, right,
    }) {
      const el = this.$refs.context.$el;
      const x = left || (right - el.offsetWidth);
      const y = top || (bottom - el.offsetHeight);
      this.$refs.context.positionMenu(y, x);
    },
    onOpen(...args) {
      this.$emit('open', ...args);
    },
  },
};
</script>

<style>
.context .btn {
  height: 40px;
  padding: 0 16px;
  margin: 0;
  cursor: pointer;

  & {
    @mixin nuggets;
  }
}
</style>
<style scoped>
.v-context.context {
  display: flex;
  width: auto;
  flex-direction: column;
  background: var(--theme-background-highlight);
  border-radius: 4px;
  border: none;
  box-shadow: 0 var(--theme-shadow-offset) 32px 16px rgba(0,0,0,0.32);
}
</style>
