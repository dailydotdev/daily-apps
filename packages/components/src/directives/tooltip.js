/* eslint-disable no-underscore-dangle, no-param-reassign */
import Vue from 'vue';
import DaTooltip from '../components/DaTooltip.vue';

const DaTooltipClass = Vue.extend(DaTooltip);
let vnode = null;
let targetElement = null;
let overTimeout = null;
let outTimeout = null;

const getPlacement = (modifiers) => {
  const keys = Object.keys(modifiers);
  if (keys.length) {
    return keys[0];
  }
  return 'top';
};

const positionTooltip = (target, tooltip, placement) => {
  const targetRect = target.getBoundingClientRect();
  const tooltipRect = tooltip.getBoundingClientRect();
  tooltip.removeAttribute('style');
  tooltip.style.position = 'absolute';
  tooltip.style.left = `${targetRect.left + (targetRect.width - tooltipRect.width) / 2}px`;
  if (placement === 'top') {
    tooltip.style.top = `${targetRect.top - tooltipRect.height - 5}px`;
  } else {
    tooltip.style.top = `${targetRect.bottom + 5}px`;
  }
};

const createTooltip = () => {
  vnode = new DaTooltipClass();
  vnode.$mount();
  document.body.appendChild(vnode.$el);
};

const showTooltip = (el, value, modifiers) => {
  targetElement = el;
  el.setAttribute('aria-describedby', 'toolip');
  vnode.content = value;
  vnode.placement = getPlacement(modifiers);
  vnode.show = true;
  setTimeout(() => positionTooltip(el, vnode.$el, vnode.placement), 10);
};

const hideTooltip = (el) => {
  targetElement = null;
  el.removeAttribute('aria-describedby');
  vnode.show = false;
};

const removeEvents = (el) => {
  if (el._tooltip.overHandler) {
    el.removeEventListener('mouseover', el._tooltip.overHandler, false);
    el.removeEventListener('mouseout', el._tooltip.outHandler, false);
    el.removeEventListener('click', el._tooltip.outHandler, false);
    el._tooltip.overHandler = null;
    el._tooltip.outHandler = null;
  }
};

const registerEvents = (el, value, modifiers, options) => {
  removeEvents(el);
  el._tooltip.overHandler = () => {
    if (overTimeout) {
      clearTimeout(overTimeout);
      overTimeout = null;
    }
    if (outTimeout && targetElement === el) {
      clearTimeout(outTimeout);
      outTimeout = null;
    }
    overTimeout = setTimeout(() => showTooltip(el, value, modifiers), options.delay.show);
  };
  el._tooltip.outHandler = () => {
    if (overTimeout) {
      clearTimeout(overTimeout);
      overTimeout = null;
    }
    if (targetElement === el) {
      if (outTimeout) {
        clearTimeout(outTimeout);
        outTimeout = null;
      }
      outTimeout = setTimeout(() => hideTooltip(el), options.delay.hide);
    }
  };
  el.addEventListener('mouseover', el._tooltip.overHandler, false);
  el.addEventListener('mouseout', el._tooltip.outHandler, false);
  el.addEventListener('click', el._tooltip.outHandler, false);
};

const directive = {
  options: {
    delay: { show: 500, hide: 100 },
  },
  bind(el, { value, modifiers }) {
    if (!vnode) {
      createTooltip();
    }
    el._tooltip = {};
    registerEvents(el, value, modifiers, directive.options);
  },
  update(el, { value, modifiers }) {
    removeEvents(el);
    registerEvents(el, value, modifiers, directive.options);
  },
  unbind(el) {
    removeEvents(el);
  },
};

export default directive;
