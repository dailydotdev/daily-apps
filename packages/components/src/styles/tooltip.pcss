.tooltip {
  display: block !important;
  z-index: 10000;
  font-size: 12px;
  padding: 4px;

  .tooltip-inner {
    background-color: var(--theme-primary);
    color: var(--theme-primary-invert);
    border-radius: 4px;
    padding: 4px 8px;
  }

  .tooltip-arrow {
    width: 0;
    height: 0;
    border-style: solid;
    position: absolute;
    margin: 2px;
    border-color: var(--theme-primary);
    z-index: 1;
  }

  &[x-placement^="top"] {
    margin-bottom: 5px;

    .tooltip-arrow {
      border-width: 5px 5px 0 5px;
      border-left-color: transparent !important;
      border-right-color: transparent !important;
      border-bottom-color: transparent !important;
      bottom: -1px;
      left: calc(50% - 5px);
      margin-top: 0;
      margin-bottom: 0;
    }
  }

  &[x-placement^="bottom"] {
    margin-top: 5px;

    .tooltip-arrow {
      border-width: 0 5px 5px 5px;
      border-left-color: transparent !important;
      border-right-color: transparent !important;
      border-top-color: transparent !important;
      top: -1px;
      left: calc(50% - 5px);
      margin-top: 0;
      margin-bottom: 0;
    }
  }

  &[x-placement^="right"] {
    margin-left: 5px;

    .tooltip-arrow {
      border-width: 5px 5px 5px 0;
      border-left-color: transparent !important;
      border-top-color: transparent !important;
      border-bottom-color: transparent !important;
      left: -1px;
      top: calc(50% - 5px);
      margin-left: 0;
      margin-right: 0;
    }
  }

  &[x-placement^="left"] {
    margin-right: 5px;

    .tooltip-arrow {
      border-width: 5px 0 5px 5px;
      border-top-color: transparent !important;
      border-right-color: transparent !important;
      border-bottom-color: transparent !important;
      right: -1px;
      top: calc(50% - 5px);
      margin-left: 0;
      margin-right: 0;
    }
  }

  &[aria-hidden='true'] {
    visibility: hidden;
    opacity: 0;
    transition: opacity .15s, visibility .15s;
  }

  &[aria-hidden='false'] {
    visibility: visible;
    opacity: 1;
    transition: opacity .15s;
  }
}
