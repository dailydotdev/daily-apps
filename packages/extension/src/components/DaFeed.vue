<template>
  <div class="feed" :class="spaciness">
    <div class="feed__insane" v-if="insaneMode">
      <template v-if="showAd">
        <da-insane-placeholder v-if="!ads.length"/>
        <da-insane-ad v-for="(item, index) in ads" :key="index" :ad="item"
                      @click="onAdClick" @impression="onAdImpression"/>
      </template>
      <da-insane-post v-for="item in posts" ref="posts" :key="item.id" :post="item"
                      @bookmark="onBookmark" @publication="onPublication" @menu="onPostMenu"
                      @click="onPostClick" :show-menu="isLoggedIn"
                      :menu-opened="selectedPostId === item.id"/>
    </div>
    <masonry class="feed__cards" :cols="cols" :gutter="gutter" :key="gutter" v-else>
      <template v-if="showAd">
        <da-card-placeholder v-if="!ads.length"/>
        <da-card-ad v-for="(item, index) in ads" :key="index" :ad="item"/>
      </template>
      <da-card-post v-for="item in posts" ref="posts" :key="item.id" :post="item"
                    @bookmark="onBookmark" @publication="onPublication" @menu="onPostMenu"
                    @click="onPostClick" :show-menu="isLoggedIn"
                    :menu-opened="selectedPostId === item.id"/>
    </masonry>
    <da-context ref="context" class="feed__context" @open="onPostMenuOpened"
                @close="selectedPostId = null">
      <button class="btn btn-menu" @click="reportPost('broken')">Broken link</button>
      <button class="btn btn-menu" @click="reportPost('nsfw')">Report NSFW</button>
      <button class="btn btn-menu" @click="hidePost">Hide post</button>
    </da-context>
  </div>
</template>

<script>
import Vue from 'vue';
import {
  mapState, mapActions, mapMutations, mapGetters,
} from 'vuex';
import VueMasonry from 'vue-masonry-css';
import { monetizationService, contentService } from '../common/services';

Vue.use(VueMasonry);

export default {
  name: 'DaFeed',
  components: {
    DaCardPost: () => import(/* webpackChunkName: "cards" */ '@daily/components/src/components/DaCardPost.vue'),
    DaCardAd: () => import(/* webpackChunkName: "cards" */ '@daily/components/src/components/DaCardAd.vue'),
    DaCardPlaceholder: () => import(/* webpackChunkName: "cards" */ '@daily/components/src/components/DaCardPlaceholder.vue'),
    DaInsanePost: () => import(/* webpackChunkName: "insane" */ '@daily/components/src/components/DaInsanePost.vue'),
    DaInsaneAd: () => import(/* webpackChunkName: "insane" */ '@daily/components/src/components/DaInsaneAd.vue'),
    DaInsanePlaceholder: () => import(/* webpackChunkName: "insane" */ '@daily/components/src/components/DaInsanePlaceholder.vue'),
    DaContext: () => import('@daily/components/src/components/DaContext.vue'),
  },
  data() {
    return {
      selectedPostId: null,
      ads: [],
      adTimeout: null,
    };
  },
  computed: {
    ...mapState('ui', ['insaneMode', 'spaciness']),
    ...mapGetters({
      posts: 'feed/feed',
      showAd: 'feed/showAd',
      isLoggedIn: 'user/isLoggedIn',
    }),
    gutter() {
      if (this.spaciness === 'roomy') {
        return 48;
      }
      if (this.spaciness === 'cozy') {
        return 56;
      }
      return 32;
    },
    cols() {
      if (this.spaciness === 'cozy') {
        return {
          default: 5,
          2221: 4,
          1919: 3,
          1617: 3,
          1315: 2,
          1061: 2,
        };
      }
      if (this.spaciness === 'roomy') {
        return {
          default: 6,
          2221: 5,
          1919: 4,
          1617: 3,
          1315: 3,
          1061: 2,
        };
      }
      return {
        default: 7,
        2221: 6,
        1919: 5,
        1617: 4,
        1315: 3,
        1061: 2,
      };
    },
  },
  watch: {
    showAd(value) {
      if (value) {
        this.ads = [];
        this.fetchAd();
      } else if (this.adTimeout) {
        clearTimeout(this.adTimeout);
      }
    },
  },
  methods: {
    onAdClick(ad) {
      ga('send', 'event', 'Ad', 'Click', ad.source);
    },

    onAdImpression(ad) {
      ga('send', 'event', 'Ad', 'Impression', ad.source);
    },

    onBookmark({ post, bookmarked }) {
      ga('send', 'event', 'Post', 'Bookmark', bookmarked ? 'Add' : 'Remove');
      this.toggleBookmarks({ id: post.id, bookmarked });
    },

    onPublication({ pub }) {
      if (this.filter && this.filter.type === 'publication' && this.filter.info.id === pub.id) return;
      ga('send', 'event', 'Post', 'Publication');
      this.setFilter({ type: 'publication', info: pub });
    },

    onPostClick(post) {
      // eslint-disable-next-line no-param-reassign
      post.read = true;
      ga('send', 'event', 'Post', 'Click', post.source);
    },

    onPostMenu({ post, event }) {
      ga('send', 'event', 'Post', 'Menu');
      this.$refs.context.open(event, post);
    },

    onPostMenuOpened(event, post) {
      const rect = event.target.getBoundingClientRect();
      this.$refs.context.positionMenu({ bottom: rect.top - 8, right: rect.right });
      this.selectedPostId = post.id;
    },

    reportPost(reason) {
      ga('send', 'event', 'Post', 'Report', reason);
      const postId = this.selectedPostId;
      this.$refs.context.close();

      // TODO: handle error
      contentService.reportPost(postId, reason)
      // eslint-disable-next-line no-console
        .catch(console.error);

      setTimeout(() => {
        this.$refs.posts.find(com => com.post.id === postId).notify('Thanks for reporting!');
        setTimeout(() => this.removePost(postId), 1000);
      }, 100);
    },

    hidePost() {
      ga('send', 'event', 'Post', 'Hide');
      const postId = this.selectedPostId;
      this.$refs.context.close();

      // TODO: handle error
      contentService.hidePost(postId)
      // eslint-disable-next-line no-console
        .catch(console.error);

      this.removePost(postId);
    },

    async fetchAd() {
      try {
        this.ads = await monetizationService.fetchAd();
        if (!this.ads.length) {
          ga('send', 'event', 'Ad', 'NotAvailable');
        }
      } catch (err) {
        // TODO: handle error
        // eslint-disable-next-line no-console
        console.error(err);
      }
      this.adTimeout = setTimeout(() => this.fetchAd(), 60000);
    },

    ...mapActions({
      setFilter: 'feed/setFilter',
    }),

    ...mapMutations({
      removePost: 'feed/removePost',
      toggleBookmarks: 'feed/toggleBookmarks',
    }),
  },

  mounted() {
    if (this.showAd) {
      this.fetchAd();
    }
  },
};
</script>

<style>
.feed__insane {
  border-radius: 8px;
  overflow: hidden;
}

.feed__cards {
  --cards-margin: 32px;
  margin: calc(var(--cards-margin) * -1) 0;

  & .card, & .card-ph {
    margin: var(--cards-margin) 0;
  }

  .roomy & {
    --cards-margin: 48px;
  }

  .cozy & {
    --cards-margin: 56px;
  }
}
</style>
