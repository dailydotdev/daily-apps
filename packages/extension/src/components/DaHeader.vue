<template>
  <header class="header">
    <a href="https://www.dailynow.co" target="_blank" class="header__logo">
      <svgicon icon="logo" class="header__logo__icon"/>
    </a>
    <div class="separator"></div>
    <da-icon-toggle class="header__theme" pressed-icon="moon" icon="sun" title="Change theme"
                    :pressed="theme > 0" @toggle="switchTheme"/>
    <da-icon-toggle class="header__insane" pressed-icon="card" icon="line" title="Change layout"
                    :pressed="insaneMode" @toggle="toggleInsane"/>
    <da-switch class="header__switch" icon="bookmark" :checked="showBookmarks"
               @toggle="toggleBookmarks"></da-switch>
    <div class="space"></div>
    <template v-if="showTopSites">
      <a v-for="(item, index) in topSites" :key="index" class="header__top-site"
         :href="item.url" :title="item.title" @mouseup="mouseUp('Top Sites')">
        <img :src="getIconUrl(item.url)" class="top-site__image"/>
      </a>
      <div class="separator"></div>
    </template>
    <a class="btn-icon" href="https://github.com/dailynowco/daily" target="_blank"
       title="Feedback us on GitHub!">
      <svgicon icon="github"/>
    </a>
    <a class="btn-icon" href="https://www.producthunt.com/posts/daily-2-0" target="_blank"
       title="Check us on Product Hunt!">
      <svgicon icon="ph"/>
    </a>
    <div class="separator"></div>
    <button class="btn-icon" title="Daily Go" @click="$emit('go')">
      <svgicon icon="mobile"/>
    </button>
    <button class="btn-icon btn-dnd" title="Do Not Disturb (DND)"
            :class="{ 'active': showDndMenu }" @click="$emit('menu', $event)">
      <svgicon icon="timer"/>
    </button>
    <button class="btn-icon btn-terminal" title="Notifications"
            :class="{ 'active': notificationsOpened }" @click="toggleNotifications">
      <svgicon icon="terminal"/>
      <span class="header__badge" v-if="showNotificationBadge"></span>
    </button>
    <button class="header__profile" v-if="isLoggedIn" @click="$emit('profile')">
      <img :src="profileImage" alt="Profile image"/>
    </button>
    <button class="btn btn-water-cheese header__sign-in" v-else
            @click="$emit('login')">
      <svgicon icon="user_daily"/>
      <span>Sign in</span>
    </button>
    <div class="instructions invert" v-if="topSitesInstructions">
      <div class="instructions__top-sites">
        <div class="header__top-site">
          <img :src="getIconUrl('https://www.google.com/')" class="top-site__image"/>
        </div>
        <div class="header__top-site">
          <img :src="getIconUrl('https://web.whatsapp.com/')" class="top-site__image"/>
        </div>
        <div class="header__top-site">
          <img :src="getIconUrl('https://www.youtube.com/')" class="top-site__image"/>
        </div>
        <div class="header__top-site">
          <img :src="getIconUrl('https://www.twitter.com/')" class="top-site__image"/>
        </div>
        <div class="header__top-site">
          <img :src="getIconUrl('https://www.facebook.com/')" class="top-site__image"/>
        </div>
      </div>
      <div class="instructions__desc">
        Should we show your recently visited sites?
      </div>
      <div class="instructions__buttons">
        <button class="btn btn-invert" @click="onToggleTopSites(false)">
          No
        </button>
        <button class="btn btn-invert" @click="onToggleTopSites(true)">
          Yes
        </button>
      </div>
    </div>
  </header>
</template>

<script>
import { mapState, mapMutations, mapGetters } from 'vuex';
import { themes } from '@daily/services';

export default {
  name: 'DaHeader',

  components: {
    DaIconToggle: () => import('@daily/components/src/components/DaIconToggle.vue'),
    DaSwitch: () => import('@daily/components/src/components/DaSwitch.vue'),
    DaContext: () => import('@daily/components/src/components/DaContext.vue'),
  },

  data() {
    return {
      topSites: [],
    };
  },

  computed: {
    ...mapState('ui', [
      'showTopSites', 'insaneMode', 'showNotificationBadge', 'notificationsOpened', 'showDndMenu',
    ]),
    ...mapState('feed', ['showBookmarks']),
    ...mapGetters('ui', ['topSitesInstructions']),
    ...mapGetters('user', ['isLoggedIn']),
    ...mapState({
      theme: state => themes.indexOf(state.ui.theme),

      notificationsOpened: state => state.ui.showNotifications,
      profileImage(state) {
        if (this.isLoggedIn) {
          return state.user.profile.image;
        }

        return '';
      },
    }),
  },

  watch: {
    async showTopSites(val) {
      if (val) {
        await this.getTopSites();
      }
    },
  },

  async mounted() {
    this.loadIcons();

    if (this.showTopSites) {
      await this.getTopSites();
    }
  },

  methods: {
    loadIcons() {
      import('@daily/components/icons/logo');
      import('@daily/components/icons/sun');
      import('@daily/components/icons/moon');
      import('@daily/components/icons/card');
      import('@daily/components/icons/line');
      import('@daily/components/icons/bookmark');
      import('@daily/components/icons/user_daily');
      import('@daily/components/icons/terminal');
      import('@daily/components/icons/mobile');
      import('@daily/components/icons/ph');
      import('@daily/components/icons/github');
      import('@daily/components/icons/timer');
    },

    async getTopSites() {
      try {
        if ('topSites' in browser) {
          this.topSites = (await browser.topSites.get()).slice(0, 5);
        }
        return [];
      } catch {
        return [];
      }
    },

    getIconUrl(url) {
      return `https://app.dailynow.co/icon?url=${encodeURIComponent(url)}&size=20`;
    },

    mouseUp(data) {
      ga('send', 'event', 'Header', 'Click', data);
    },

    switchTheme(pressed) {
      const newTheme = pressed ? themes[1] : themes[0];
      this.$store.dispatch('ui/setTheme', newTheme);
      ga('send', 'event', 'Header', 'Theme', this.theme);
    },

    toggleInsane(pressed) {
      this.$store.commit('ui/setInsaneMode', pressed);
      ga('send', 'event', 'Header', 'Insane', pressed);
    },

    toggleBookmarks(pressed) {
      this.$store.dispatch('feed/setShowBookmarks', pressed);
      ga('send', 'event', 'Header', 'Bookmarks', pressed);
    },

    toggleNotifications() {
      ga('send', 'event', 'Header', 'Terminal', !this.notificationsOpened);
      if (this.notificationsOpened) {
        this.hideNotifications();
      } else {
        this.showNotifications();
      }
    },

    async onToggleTopSites(val) {
      ga('send', 'event', 'Instructions', 'Click', 'Top Sites');
      // TODO: handle error
      this.$store.commit('ui/nextInstruction');
      await this.$store.dispatch('ui/setShowTopSites', val);
    },

    ...mapMutations({
      hideNotifications: 'ui/hideNotifications',
      showNotifications: 'ui/showNotifications',
    }),
  },
};
</script>

<style>
.header {
  position: relative;
  display: flex;
  width: 100%;
  height: 48px;
  flex-direction: row;
  align-items: center;
  padding: 0 8px;
  background: var(--theme-background-highlight);
  border-bottom: 1px solid var(--theme-separator);
  z-index: 30;
  contain: layout size;

  & > * {
    margin: 0 4px;
  }

  & > .btn-icon {
    margin-left: 2px;
    margin-right: 2px;
  }

  & .header__logo {
    background: none;
    border: none;
    padding: 0;
  }

  & .header__logo__icon {
    width: 40px;
    height: 40px;
    color: var(--theme-primary);
  }

  & .separator {
    margin-left: 8px;
    margin-right: 8px;
  }

  & .header__switch {
    width: var(--da-switch-width);
    height: var(--da-switch-height);
    margin-left: 8px;

    --da-switch-checked-color: var(--color-burger-60);
    --da-switch-checked-background: var(--color-burger-90);

    &:hover {
      --da-switch-checked-color: var(--color-burger-50);
    }

    .bright & {
      --da-switch-checked-background: var(--color-burger-30);

      &:hover {
        --da-switch-checked-color: var(--color-burger-70);
      }
    }
  }

  & .header__sign-in {
    margin-left: 8px;
    margin-right: 0;
  }

  & .header__top-site {
    width: 24px;
    height: 24px;
    border-radius: 50%;
    border: 1px solid var(--color-pepper-70);
    overflow: hidden;
  }

  & .top-site__image {
    width: 23px;
    height: 23px;
    background: var(--color-salt-10);
  }

  & .space {
    flex: 1;
  }

  & .header__profile {
    width: 30px;
    height: 30px;
    padding: 0;
    margin: 0 8px 0 14px;
    border-radius: 4px;
    overflow: hidden;
    background: none;
    border: none;
    cursor: pointer;

    & img {
      width: 100%;
      height: 100%;
    }
  }

  & .instructions {
    top: 0;
    right: 190px;
    width: 222px;
  }

  & .instructions__desc {
    text-align: center;
  }
}

.header__badge {
  position: absolute;
  left: 17px;
  bottom: 16px;
  width: 10px;
  height: 10px;
  padding: 2px;
  background: var(--theme-background-highlight);
  border-radius: 100%;

  &:before {
    content: '';
    display: block;
    width: 100%;
    height: 100%;
    background: var(--color-water-30);
    border-radius: 100%;
  }
}

.instructions__top-sites {
  display: flex;
  flex-direction: row;
  margin: 0 -4px 12px;

  & .header__top-site {
    margin: 0 4px;
    cursor: default;
  }
}

.instructions__buttons {
  display: flex;
  flex-direction: row;
  align-self: stretch;
  margin-top: 8px;

  & .btn {
    margin: 0 4px;
    flex: 1;
    justify-content: center;
  }
}

@media (min-width: 1071px) {
  .header .header__switch {
    position: absolute;
    left: 0;
    right: 0;
    top: 0;
    bottom: 0;
    margin: auto;
  }
}
</style>
