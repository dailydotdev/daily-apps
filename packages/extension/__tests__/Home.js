import { mount, createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';
import svgicon from 'vue-svgicon';
import tooltip from '@daily/components/src/directives/tooltip';
import DaHome from '../src/routes/Home.vue';
import DaContext from '../../components/src/components/DaContext.vue';
import DaHeader from '../src/components/DaHeader.vue';
import DaSidebar from '../src/components/DaSidebar.vue';
import DaFeed from '../src/components/DaFeed.vue';

const localVue = createLocalVue();

localVue.use(Vuex);
localVue.use(svgicon);
localVue.directive('tooltip', tooltip);
localVue.component('da-context', DaContext);
localVue.component('da-header', DaHeader);
localVue.component('da-sidebar', DaSidebar);
localVue.component('da-feed', DaFeed);

let feed;
let ui;
let user;
let store;

beforeEach(() => {
  window.ga = () => {
  };

  feed = {
    namespaced: true,
    state: {
      publications: [],
      posts: [],
    },
    mutations: {
      setPublications: jest.fn()
    },
    getters: {
      feed: state => state['posts'],
      showAd: jest.fn(),
    }
  };

  ui = {
    namespaced: true,
    state: {
      insaneMode: false,
    },
    mutations: {
      setDndModeTime: jest.fn(),
      setShowDndMenu: jest.fn(),
    },
    getters: {
      topSitesInstructions: jest.fn(),
      sidebarInstructions: jest.fn(),
      showReadyModal: jest.fn(),
      dndMode: jest.fn(),
      dndModeTime: jest.fn(),
    },
  };

  user = {
    namespaced: true,
    state: {
      profile: null,
    },
    getters: {
      isLoggedIn: state => !!state.profile,
    },
    actions: {
      refreshToken: jest.fn()
    }
  };

  store = new Vuex.Store({
    modules: { feed, ui, user },
  });

  window.requestIdleCallback = jest.fn();
  window.IntersectionObserver = jest.fn();
});

it('should commit "setDndModeTime" when "For 1 Hour" or "Until Tomorrow" is clicked', () => {
  const wrapper = mount(DaHome, { store, localVue });
  wrapper.find('.dnd-context .btn-menu').trigger('click');
  expect(ui.mutations.setDndModeTime).toBeCalledWith(expect.anything(), expect.any(Number));
});
