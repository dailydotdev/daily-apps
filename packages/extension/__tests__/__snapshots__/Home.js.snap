// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should show banner when data is available 1`] = `
<div class="banner gradient-bacon-onion"><span class="lil2 title">Title</span> <span class="micro1">Sub</span> <a href="https://daily.dev" target="_blank" class="btn cta">Click here</a> <button class="btn-icon btn-small"><svg version="1.1" viewBox="0 0 24 24" class="svg-icon svg-fill">
      <path pid="0" d="M12 9.44l3.91-3.91a1.81 1.81 0 012.56 2.56L14.56 12l3.91 3.91a1.81 1.81 0 01-2.56 2.56L12 14.56l-3.91 3.91a1.81 1.81 0 11-2.56-2.56L9.44 12 5.53 8.09a1.81 1.81 0 112.56-2.56L12 9.44z" _fill="#FFF" fill-rule="evenodd"></path>
    </svg></button></div>
`;
